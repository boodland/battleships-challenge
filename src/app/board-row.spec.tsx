import { render } from '@testing-library/react';

import BoardRow from './board-row';

vi.mock('./utils', () => {
  return { hasShip: () => true, columns: [1, 2, 3, 4, 5] };
});
vi.mock('./board-cell', () => ({ default: (props: unknown) => `MockedBoardCell ${JSON.stringify(props)}` }));

const onCellClick = (position: number) => null;

describe('BoardRow', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <BoardRow
        startPosition={0}
        ships={[]}
        shots={[]}
        showShips={false}
        onCellClick={onCellClick}
      />
    );
    expect(baseElement).toBeTruthy();
  });

  it('should display one BoardCell per column', () => {
    const { container } = render(
      <BoardRow
        startPosition={0}
        ships={[]}
        shots={[]}
        showShips={false}
        onCellClick={onCellClick}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('should pass the correct params to BoardCells', () => {
    const { container } = render(
      <BoardRow
        startPosition={0}
        ships={[]}
        shots={[false, true, false, true, true]}
        showShips={true}
        onCellClick={onCellClick}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
