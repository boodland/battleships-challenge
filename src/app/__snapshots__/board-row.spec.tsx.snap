// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BoardRow > should display one BoardCell per column 1`] = `
<div>
  MockedBoardCell {"hasShip":true,"showShip":false}
  MockedBoardCell {"hasShip":true,"showShip":false}
  MockedBoardCell {"hasShip":true,"showShip":false}
  MockedBoardCell {"hasShip":true,"showShip":false}
  MockedBoardCell {"hasShip":true,"showShip":false}
</div>
`;

exports[`BoardRow > should pass the correct params to BoardCells 1`] = `
<div>
  MockedBoardCell {"shotted":false,"hasShip":true,"showShip":true}
  MockedBoardCell {"shotted":true,"hasShip":true,"showShip":true}
  MockedBoardCell {"shotted":false,"hasShip":true,"showShip":true}
  MockedBoardCell {"shotted":true,"hasShip":true,"showShip":true}
  MockedBoardCell {"shotted":true,"hasShip":true,"showShip":true}
</div>
`;
